File: platform/android/MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/maps/AttributionDialogManagerTest.java
package com.mapbox.mapboxsdk.maps;

import android.content.Context;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertNotNull;
import static org.mockito.Mockito.mock;

@RunWith(RobolectricTestRunner.class)
public class AttributionDialogManagerTest {
  @InjectMocks
  Context context = mock(Context.class);

  @InjectMocks
  MapboxMap mapboxMap = mock(MapboxMap.class);

  @InjectMocks
  Style style = mock(Style.class);

  private AttributionDialogManager attributionDialogManager;

  @Before
  public void beforeTest() {
    attributionDialogManager = new AttributionDialogManager(context, mapboxMap);
  }

  @Test
  public void testSanity() {
    assertNotNull("AttributionDialogManager should not be null", attributionDialogManager);
  }
}

File: platform/android/MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/maps/AttributionDialogManagerTest.kt
package com.mapbox.mapboxsdk.maps

import android.content.Context
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.InjectMocks
import org.mockito.Mockito
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class AttributionDialogManagerTest {
    @InjectMocks
    var context = Mockito.mock(
        Context::class.java
    )

    @InjectMocks
    var mapboxMap = Mockito.mock(MapboxMap::class.java)

    @InjectMocks
    var style = Mockito.mock(
        Style::class.java
    )
    private var attributionDialogManager: AttributionDialogManager? = null

    @Before
    fun beforeTest() {
        attributionDialogManager = AttributionDialogManager(context, mapboxMap)
    }

    @Test
    fun testSanity() {
        Assert.assertNotNull(
            "AttributionDialogManager should not be null",
            attributionDialogManager
        )
    }
}
